File: LANCommander.SDK/Clients/LibraryClient.cs
using LANCommander.SDK.Models;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using LANCommander.SDK.Factories;

namespace LANCommander.SDK.Services
{
    public class LibraryClient(ApiRequestFactory apiRequestFactory)
    {
        public async Task<IEnumerable<EntityReference>> GetAsync()
        {
            var results = await apiRequestFactory
                .Create()
                .UseAuthenticationToken()
                .UseVersioning()
                .UseRoute("/api/Library")
                .GetAsync<IEnumerable<EntityReference>>();

            return results ?? [];
        }

        public async Task<bool> AddToLibrary(Guid gameId)
        {
            return await apiRequestFactory
                .Create()
                .UseAuthenticationToken()
                .UseVersioning()
                .UseRoute($"/api/Library/AddToLibrary/{gameId}")
                .PostAsync<bool>();
        }

        public async Task<bool> RemoveFromLibrary(Guid gameId)
        {
            return await apiRequestFactory
                .Create()
                .UseAuthenticationToken()
                .UseVersioning()
                .UseRoute($"/api/Library/RemoveFromLibrary/{gameId}")
                .PostAsync<bool>();
        }

        public async Task<bool> RemoveFromLibrary(Guid gameId, Guid[] addonIds)
        {
            return await apiRequestFactory
                .Create()
                .UseAuthenticationToken()
                .UseVersioning()
                .UseRoute($"/api/Library/RemoveFromLibrary/{gameId}/addons")
                .AddBody(new GenericGuidsRequest
                {
                    Guids = addonIds
                })
                .PostAsync<bool>();
        }
    }
}

File: LANCommander.Launcher.Win9x/src/library_client.cpp
#include "library_client.h"

#include <sstream>

bool LibraryClient::AddToLibrary(const std::string& gameId, std::string* errorOut)
{
    HttpResponse resp = m_http.PostJson(
        "/api/Library/AddToLibrary/" + gameId, std::string());
    if (resp.ok()) return true;
    if (errorOut)
    {
        std::ostringstream e;
        e << "AddToLibrary failed (HTTP " << resp.status << ")";
        *errorOut = e.str();
    }
    return false;
}

bool LibraryClient::RemoveFromLibrary(const std::string& gameId, std::string* errorOut)
{
    HttpResponse resp = m_http.PostJson(
        "/api/Library/RemoveFromLibrary/" + gameId, std::string());
    if (resp.ok()) return true;
    if (errorOut)
    {
        std::ostringstream e;
        e << "RemoveFromLibrary failed (HTTP " << resp.status << ")";
        *errorOut = e.str();
    }
    return false;
}
